tsdb/engine/tsm1/engine.go

```diff
@@ -653,6 +653,162 @@ func (e *Engine) Backup(w io.Writer, basePath string, since time.Time) error {
 	return nil
 }
 
+// TODO(Adam): this is copied from another branch, needs more work.
+func (e *Engine) Export(w io.Writer, basePath string, start time.Time, end time.Time) error {
+	path, err := e.CreateSnapshot()
+	if err != nil {
+		return err
+	}
+
+	if err := e.index.SnapshotTo(path); err != nil {
+		return err
+	}
+
+	tw := tar.NewWriter(w)
+	defer tw.Close()
+
+	// Remove the temporary snapshot dir
+	defer os.RemoveAll(path)
+
+	// Recursively read all files from path.
+	files, err := readDir(path, "")
+	if err != nil {
+		return err
+	}
+
+	// filtered is different here, these are files we need to trim based on timestamps.
+	// whole is an optimisation: if a TSM is fully contained in the range, then we keep the whole thing.
+	var filtered []string
+	var whole []string
+	for _, file := range files {
+		if !strings.HasSuffix(file, ".tsm") {
+			continue
+		}
+
+		var tombstonePath string
+		f, err := os.Open(filepath.Join(path, file))
+		if err != nil {
+			return err
+		}
+		r, err := NewTSMReader(f)
```

Comment:
Opening a TSMReader multiple times is kind of expensive.  It's done here (and thrown away), then again here: https://github.com/influxdata/influxdb/pull/8941/files#diff-82e4e65bd72f00208a49c4797d466ecdR768

It would better to only have to open each file once.